packages/marshaler.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package packages
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+type MarshallerOption func(packages *Packages) error
+
+func MarshalJSON(pkgs *Packages) ([]byte, error) {
+	return json.MarshalIndent(pkgs, " ", " ")
+}
+
+func UnmarshalJSON(content []byte, pkgs *Packages, options ...MarshallerOption) error {
```

Comment:
Could `Marshal` and `Unmarshal` have been implemented as methods of the `Packages` type, so it implements `Un`/`marshaller`?

Well, I guess we need it this way because of the options?

Reply:
If we decide to drop all options, then yes, but it would be hackish too. Here is [an article](http://choly.ca/post/go-json-marshalling/) about the approach.

Reply:
Sounds good, we can keep it this way.